### DISCUSSION TITLE
Is there a build that can be included in browser

### DISCUSSION THREAD

**Discussion by taypo** [ueberdosis/tiptap#744] [General Questions]
Hi,

Is there a build of tiptap (with its deps) that I can include in the browser, and use it without npm/webpack etc. Just like I can do with [Vue.js](https://vuejs.org/v2/guide/#Getting-Started) itself.

**Comment by davidkhess** [comment]
I wanted to followup on this question also. I'm using Vue 2 but *not* vue-cli. I'm using native browser imports and packing with Rollup using SystemJS format. So, I need either a non-module, browser package of tiptap or an ESM format where bare imports have been converted to paths.

The Skypack CDN instructions almost work for my use case, but it loads in an additional copy of Vue 2 itself and multiple versions of prosemirror-model (apparently due to ambiguous pinning in package.json dependencies?) which results in editor errors. There doesn't seem to be a way to get Skypack to not do this.

I also tried the UMD dists but they don't seem to work - it looks like the package registration doesn't work correctly and a reference to `core` on the Window object fails. Seems like the packages are registering as their full NPM package names instead of things like just `core`.

Does anybody have any suggestions on how to deal with this? Can support for non-bare, native browser imports or a non-dependency pulling browser dist be added?

As a point of reference Vue is providing an esm browser dist: `vue.esm.browser.js`

At the moment, this is looking like a deal-breaker for using tiptap.

**Comment by davidkhess** [comment]
I came up with a solution using rollup and the node-resolve rollup plugin. I put this code in a file called rollup.js:

```
import { Editor, EditorContent } from "@tiptap/vue-2"
import StarterKit from "@tiptap/starter-kit"
export { Editor, EditorContent, StarterKit }
```

Then bundle it with rollup using a config like this:

```
import { nodeResolve } from "@rollup/plugin-node-resolve";
import replace from "@rollup/plugin-replace";
import { terser } from 'rollup-plugin-terser';

export default {
    input: "app/frontend/src/rollup/rollup.js",
    output: [
        {
            file: "app/static/dist/javascript/rollup.js",
            format: "iife",
            name: "window",
            extend: true
        },
        {
            file: "app/static/dist/javascript/rollup.min.js",
            format: "iife",
            name: "window",
            extend: true,
            plugins: [ terser() ]
        }
    ],
    plugins: [
        replace({
            "process.env.NODE_ENV": "'production'"
        }),
        nodeResolve()
    ]
};
```

It will resolve the bare module references in the tiptap and prose mirror modules and then make the exported symbols available on Window when loaded via `script` tag. You can of course use this same approach for any other modules that support ES module importing and also use bare module references.

Note, you could also bundle to ES module format and do a relative browser module import (I couldn't in this case because I was also including some other packages that other non-ES modules needed to see as globals on Window - ES modules load `defer` which caused me timing issues).
